In [1]:
import requests
from bs4 import BeautifulSoup
import pandas as pd
import requests
import re

import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)
pd.set_option('display.width', None)
pd.set_option('display.max_colwidth', None)

search_filmaffinity: I wanted to automatize the search of the link of each movie in the list. This functions takes the "title" in spanish of the movie and search the filmaffinity database to get to the main page of each movie. I simulate a real user using the search function, like search in the main bar: "El señor de los anillos". It returns the title and the link of the main page.

We have an inside function to get matches when i search in the database. If you search "El señor de los anillos" you get a few matches, i get a function to pick the one with the exact main title. if not, it returns "couldnt find", "couldnt find". 

In [3]:
## title must be in spanish, so we get matches in the filmaffinity database.
## this function works 95% of the time. Sometimes it returns me an incorrect movie becouse of the title_main in title_scraping.
## must be a better way to do it. i think it would work if i apply strip to title_scraping. Sometimes filmaff puts a blank
# space to the end of the title, givine me something like: "Origen ".

    
def search_filmaffinity_movie_href(title):
    
    try:
        
        def search_filmaffinity_scraping(soup): ## this function search in fa search bar, and iterates the results to get matches
            products = soup.find_all("div", attrs = {"class":"mc-title"})
            for i in products:
                href = i.find("a").get("href")
                title_scraping = i.find("a").get("title").lower().strip()
                if title_main == title_scraping: ## if title == movie match, give me the movie page link.
                    return title_scraping, href
                elif title_main in title_scraping:## elif exists for exceptions, maybe.
                    return title_scraping, href

            return "Couldnt find", "Couldn't find"

        title_main = title.lower()
        title_cleaned_search = title_main.replace(" ","+")
        search_url_filmaffinity = "https://www.filmaffinity.com/es/search.php?stext="+title_cleaned_search ## generates link search
        res = requests.get(search_url_filmaffinity)
        html = res.content
        soup = BeautifulSoup(html, "html.parser")
        content = soup.find_all("div", attrs = {"class":"meta content"}) 
        title_scraping, href = search_filmaffinity_scraping(soup) ## aplies function
        if href == "Couldn't find": ## if we don't find match
            content = soup.find_all("dd", attrs = {"itemprop":"description"}) 
            if content: 
            #this if function is important. Sometimes filmaff don't find a match but returns the main page of the movie if the search
            # is exactly like the name. so when this happens, i check if content(this is the way i check im in the page of the movie
            # i want to get info) exsits, then returns the main url, which is the movie info, and the name. GGWP.
                return search_url_filmaffinity  
        else:
            return href 
    
    except:
        return np.nan




It gets a string of the link of the main page in filmaffinity and returns the stars as a float.

In [4]:
def get_note_filmaffinity(href):
    try:
        res = requests.get(href)
        html = res.content
        soup = BeautifulSoup(html, "html.parser")
        content = soup.find_all("div", attrs = {"itemprop":"ratingValue"})
        pov = content[0].get_text()
        pov = pov.replace(",",".")
        note = float(pov.strip())
        if not note:
            return "No note"
        return note
    except:
        return np.nan

It gets a string of the main page in filmaffinity and returns the number of voters, as a integer.

In [5]:
def get_votes(href):
    try:
        res = requests.get(href)
        html = res.content
        soup = BeautifulSoup(html, "html.parser")
        content = soup.find_all("div", attrs = {"id":"movie-count-rat"})
        pov = content[0].get_text()
        votes = int(pov.strip().strip('\nvotos').replace(".",""))
        return votes
    
    except:
        return np.nan

In [6]:
print(get_votes("https://www.filmaffinity.com/es/film113180.html"))

1461


Gets the original name of the main page of the ff database as a string

In [7]:
def get_name(href):
    try:
        res = requests.get(href)
        html = res.content
        soup = BeautifulSoup(html, "html.parser")
        content = soup.find_all("div", attrs = {"id":"left-column"})
        content = content[0].find("dd")
        original_name = content.get_text().strip()
        return original_name
    
    except:
        return np.nan

In [8]:
print(get_note_filmaffinity("https://www.filmaffinity.com/es/film113180.html"))

6.0


This is my main imdb database, extracted from kaggle

In [9]:
GSAF = r"C:\Users\lenovo\Desktop\Ironhack\Second project - IMDB-FA\Src\imdb_top_1000.csv"

df = pd.read_csv(GSAF, encoding = "ISO-8859-1")

In [10]:
df.head(1)

Poster_Link  \
0  https://m.media-amazon.com/images/M/MV5BMDFkYTc0MGEtZmNhMC00ZDIzLWFmNTEtODM1ZmRlYWMwMWFmXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_UX67_CR0,0,67,98_AL_.jpg   

               Series_Title Released_Year Certificate  Runtime  Genre  \
0  The Shawshank Redemption          1994           A  142 min  Drama   

   IMDB_Rating  \
0          9.3   

                                                                                                                 Overview  \
0  Two imprisoned men bond over a number of years, finding solace and eventual redemption through acts of common decency.   

   Meta_score        Director        Star1           Star2       Star3  \
0        80.0  Frank Darabont  Tim Robbins  Morgan Freeman  Bob Gunton   

            Star4  No_of_Votes       Gross  
0  William Sadler      2343110  28,341,469

This database i use it to get the titleID of the movie so i can find the title of the movie, which i have in english, in spanish.

In [11]:
GSAF = r"C:\Users\lenovo\Desktop\Ironhack\Second project - IMDB-FA\Src\data.tsv"

df_originalnames = pd.read_csv(GSAF, sep='\t')

C:\Users\lenovo\AppData\Local\Temp\ipykernel_3092\1744990979.py:3: DtypeWarning: Columns (7) have mixed types. Specify dtype option on import or set low_memory=False.
  df_originalnames = pd.read_csv(GSAF, sep='\t')


In [12]:
df_originalnames.sample(10) ## we have titleID

titleId  ordering                              title region  \
11690292   tt4584870         2  17 अप्रैल 2015 को प्रसारित एपिसोड     IN   
4373503   tt10818710         7                       Folge #1.555     DE   
6787400    tt1619582         1                            Ballads     US   
17890057   tt9101096         3                        Folge #1.22     DE   
13070998   tt5627304         1                     Episódio #1.14     PT   
394572     tt0065795         5             Dziewczyna z Las Vegas     PL   
2237817    tt0847096         1                 Sorority Splash #4     US   
4906369    tt1108913         2                         エピソード #1.3     JP   
13149164   tt5694810         7                     Episodio #1.20     IT   
2471748    tt0935420         2                     एपिसोड #1.2981     IN   

         language types attributes isOriginalTitle  
11690292       hi    \N         \N               0  
4373503        de    \N         \N               0  
6787400        \N    \N         \N               0  
17890057       de    \N         \N               0  
13070998       pt    \N         \N               0  
394572         \N    \N         \N               0  
2237817        \N    \N         \N               0  
4906369        ja    \N         \N               0  
13149164       it    \N         \N               0  
2471748        hi    \N         \N               0

In [13]:
## i merge it with my main database

In [14]:
df_number_imdb = df.merge(df_originalnames, left_on='Series_Title', right_on='title')[['Series_Title', 'titleId', 'IMDB_Rating', 'Director', "No_of_Votes"]]

In [15]:
df_number_imdb.head(20)

df_number_imdb = df_number_imdb.drop_duplicates(subset=["Series_Title"], keep=False)

In [16]:
df_number_imdb

Series_Title  \
48                                                   The Godfather: Part II   
95                                                         Schindler's List   
183                                         Il buono, il brutto, il cattivo   
210                          Star Wars: Episode V - The Empire Strikes Back   
237                                                            Gisaengchung   
253                                                          Cidade de Deus   
256                                           Sen to Chihiro no kamikakushi   
264                                                          The Green Mile   
274                                                The Silence of the Lambs   
293                                                    Shichinin no samurai   
395                                                        The Intouchables   
398                                                            The Prestige   
472                                                      American History X   
517                                                   Nuovo Cinema Paradiso   
668                                               Ayla: The Daughter of War   
669                                                            Vikram Vedha   
671                                                          Kimi no na wa.   
678                                       Spider-Man: Into the Spider-Verse   
685                                                       Avengers: Endgame   
703                                                  Avengers: Infinity War   
757                                                                3 Idiots   
761                                                        Taare Zameen Par   
774                                                                Oldeuboi   
807                                                           Mononoke-hime   
809                                             Once Upon a Time in America   
887    Dr. Strangelove or: How I Learned to Stop Worrying and Love the Bomb   
932                                                      The Great Dictator   
952                                                                 Tumbbad   
954                                                               Andhadhun   
969                                                 Jodaeiye Nader az Simin   
971                                                               Incendies   
977                                                          Babam ve Oglum   
1043                                                      Good Will Hunting   
1052                                                      Bacheha-Ye aseman   
1082                                                      Full Metal Jacket   
1098                                                           Idi i smotri   
1260                                                     A Clockwork Orange   
1265                                                  2001: A Space Odyssey   
1325                                                     North by Northwest   
1362                                                    Singin' in the Rain   
1380                                                    Ladri di biciclette   
1521                                                             Chhichhore   
1528                                               Uri: The Surgical Strike   
1533                                                       K.G.F: Chapter 1   
1536                                                             Green Book   
1549                              Three Billboards Outside Ebbing, Missouri   
1557                                                                 Talvar   
1559                                            Baahubali 2: The Conclusion   
1594                                                            Mandariinid   
1596                                                     Gangs of Wasseypur   
1607                              

With this function, i get the original imdb id for the movie and i search i get the soup. Funny is that my browser gets me to the SPANISH version of the movie, getting the spanish name with the titled. It returns a string.

In [17]:
def get_spanish_name(titled):
    try:
        href = "https://www.imdb.com/title/"+titled
        res = requests.get(href)
        html = res.content
        soup = BeautifulSoup(html, "html.parser")
        content = soup.find("title").get_text()
        content = content.strip("(")
        content = content.split("(")[0].strip()
        return content
    except:
        return np.nan


In [18]:
df_number_imdb["Spanish name"] = df_number_imdb["titleId"].apply(get_spanish_name)

In [19]:
df_number_imdb

Series_Title  \
48                                                   The Godfather: Part II   
95                                                         Schindler's List   
183                                         Il buono, il brutto, il cattivo   
210                          Star Wars: Episode V - The Empire Strikes Back   
237                                                            Gisaengchung   
253                                                          Cidade de Deus   
256                                           Sen to Chihiro no kamikakushi   
264                                                          The Green Mile   
274                                                The Silence of the Lambs   
293                                                    Shichinin no samurai   
395                                                        The Intouchables   
398                                                            The Prestige   
472                                                      American History X   
517                                                   Nuovo Cinema Paradiso   
668                                               Ayla: The Daughter of War   
669                                                            Vikram Vedha   
671                                                          Kimi no na wa.   
678                                       Spider-Man: Into the Spider-Verse   
685                                                       Avengers: Endgame   
703                                                  Avengers: Infinity War   
757                                                                3 Idiots   
761                                                        Taare Zameen Par   
774                                                                Oldeuboi   
807                                                           Mononoke-hime   
809                                             Once Upon a Time in America   
887    Dr. Strangelove or: How I Learned to Stop Worrying and Love the Bomb   
932                                                      The Great Dictator   
952                                                                 Tumbbad   
954                                                               Andhadhun   
969                                                 Jodaeiye Nader az Simin   
971                                                               Incendies   
977                                                          Babam ve Oglum   
1043                                                      Good Will Hunting   
1052                                                      Bacheha-Ye aseman   
1082                                                      Full Metal Jacket   
1098                                                           Idi i smotri   
1260                                                     A Clockwork Orange   
1265                                                  2001: A Space Odyssey   
1325                                                     North by Northwest   
1362                                                    Singin' in the Rain   
1380                                                    Ladri di biciclette   
1521                                                             Chhichhore   
1528                                               Uri: The Surgical Strike   
1533                                                       K.G.F: Chapter 1   
1536                                                             Green Book   
1549                              Three Billboards Outside Ebbing, Missouri   
1557                                                                 Talvar   
1559                                            Baahubali 2: The Conclusion   
1594                                                            Mandariinid   
1596                                                     Gangs of Wasseypur   
1607                              

In [20]:

df_number_imdb.columns = ['Series_Title_1', 'titleId', 'IMDB_Rating_1', 'Director_1', "No_of_Votes_1", "Spanish name"]

In [21]:
df.head(1)

Poster_Link  \
0  https://m.media-amazon.com/images/M/MV5BMDFkYTc0MGEtZmNhMC00ZDIzLWFmNTEtODM1ZmRlYWMwMWFmXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_UX67_CR0,0,67,98_AL_.jpg   

               Series_Title Released_Year Certificate  Runtime  Genre  \
0  The Shawshank Redemption          1994           A  142 min  Drama   

   IMDB_Rating  \
0          9.3   

                                                                                                                 Overview  \
0  Two imprisoned men bond over a number of years, finding solace and eventual redemption through acts of common decency.   

   Meta_score        Director        Star1           Star2       Star3  \
0        80.0  Frank Darabont  Tim Robbins  Morgan Freeman  Bob Gunton   

            Star4  No_of_Votes       Gross  
0  William Sadler      2343110  28,341,469

In [22]:
df_before_filmaff = df.merge(df_number_imdb, left_on='Series_Title', right_on='Series_Title_1')[['Series_Title', "Released_Year", "Runtime", "Genre", "IMDB_Rating", "Director", "Star1", "Star2", "Star3", "Star4","No_of_Votes","Gross", "Spanish name"]]

In [23]:
df_before_filmaff

Series_Title  \
0                                                  The Godfather: Part II   
1                                                        Schindler's List   
2                                         Il buono, il brutto, il cattivo   
3                          Star Wars: Episode V - The Empire Strikes Back   
4                                                            Gisaengchung   
5                                                          Cidade de Deus   
6                                           Sen to Chihiro no kamikakushi   
7                                                          The Green Mile   
8                                                The Silence of the Lambs   
9                                                    Shichinin no samurai   
10                                                       The Intouchables   
11                                                           The Prestige   
12                                                     American History X   
13                                                  Nuovo Cinema Paradiso   
14                                              Ayla: The Daughter of War   
15                                                           Vikram Vedha   
16                                                         Kimi no na wa.   
17                                      Spider-Man: Into the Spider-Verse   
18                                                      Avengers: Endgame   
19                                                 Avengers: Infinity War   
20                                                               3 Idiots   
21                                                       Taare Zameen Par   
22                                                               Oldeuboi   
23                                                          Mononoke-hime   
24                                            Once Upon a Time in America   
25   Dr. Strangelove or: How I Learned to Stop Worrying and Love the Bomb   
26                                                     The Great Dictator   
27                                                                Tumbbad   
28                                                              Andhadhun   
29                                                Jodaeiye Nader az Simin   
30                                                              Incendies   
31                                                         Babam ve Oglum   
32                                                      Good Will Hunting   
33                                                      Bacheha-Ye aseman   
34                                                      Full Metal Jacket   
35                                                           Idi i smotri   
36                                                     A Clockwork Orange   
37                                                  2001: A Space Odyssey   
38                                                     North by Northwest   
39                                                    Singin' in the Rain   
40                                                    Ladri di biciclette   
41                                                             Chhichhore   
42                                               Uri: The Surgical Strike   
43                                                       K.G.F: Chapter 1   
44                                                             Green Book   
45                              Three Billboards Outside Ebbing, Missouri   
46                                                                 Talvar   
47                                            Baahubali 2: The Conclusion   
48                                                            Mandariinid   
49                                                     Gangs of Wasseypur   
50                                                       Paan Singh Tomar   
51                                                 El sec

In [24]:
df_before_filmaff["Filmaffinity link"] = df_before_filmaff["Spanish name"].apply(search_filmaffinity_movie_href)

In [25]:
df_before_filmaff.head(5)

Series_Title Released_Year  Runtime  \
0                          The Godfather: Part II          1974  202 min   
1                                Schindler's List          1993  195 min   
2                 Il buono, il brutto, il cattivo          1966  161 min   
3  Star Wars: Episode V - The Empire Strikes Back          1980  124 min   
4                                    Gisaengchung          2019  132 min   

                        Genre  IMDB_Rating              Director  \
0                Crime, Drama          9.0  Francis Ford Coppola   
1   Biography, Drama, History          8.9      Steven Spielberg   
2                     Western          8.8          Sergio Leone   
3  Action, Adventure, Fantasy          8.7        Irvin Kershner   
4     Comedy, Drama, Thriller          8.6          Bong Joon Ho   

            Star1           Star2          Star3               Star4  \
0       Al Pacino  Robert De Niro  Robert Duvall        Diane Keaton   
1     Liam Neeson   Ralph Fiennes   Ben Kingsley    Caroline Goodall   
2  Clint Eastwood     Eli Wallach  Lee Van Cleef       Aldo Giuffrè   
3     Mark Hamill   Harrison Ford  Carrie Fisher  Billy Dee Williams   
4    Kang-ho Song    Lee Sun-kyun  Cho Yeo-jeong        Choi Woo-sik   

   No_of_Votes        Gross                Spanish name  \
0      1129952   57,300,000        El padrino: Parte II   
1      1213505   96,898,818       La lista de Schindler   
2       688390    6,100,000  El bueno, el feo y el malo   
3      1159315  290,475,067      El Imperio contraataca   
4       552778   53,367,844                   Parásitos   

                                                             Filmaffinity link  
0        https://www.filmaffinity.com/es/search.php?stext=el+padrino:+parte+ii  
1       https://www.filmaffinity.com/es/search.php?stext=la+lista+de+schindler  
2  https://www.filmaffinity.com/es/search.php?stext=el+bueno,+el+feo+y+el+malo  
3                              https://www.filmaffinity.com/es/film605090.html  
4                              https://www.filmaffinity.com/es/film520465.html

Now that i have the href, lets see how many nan values it returned me:

In [26]:
print(len(df_before_filmaff[df_before_filmaff['Filmaffinity link'].isna()]))
df_before_filmaff[df_before_filmaff['Filmaffinity link'].isna()]

13


Series_Title  \
8                                                The Silence of the Lambs   
25   Dr. Strangelove or: How I Learned to Stop Worrying and Love the Bomb   
33                                                      Bacheha-Ye aseman   
98                                                   Pink Floyd: The Wall   
116                                          Das Cabinet des Dr. Caligari   
146                                                    Dances with Wolves   
152                                       Who's Afraid of Virginia Woolf?   
240                                           What's Eating Gilbert Grape   
243                                          Batman: Mask of the Phantasm   
244                                            Back to the Future Part II   
286                                   Interstate 60: Episodes of the Road   
311                                                        La Vie En Rose   
326                                                              Dead Man   

    Released_Year  Runtime                      Genre  IMDB_Rating  \
8            1991  118 min     Crime, Drama, Thriller          8.6   
25           1964   95 min                     Comedy          8.4   
33           1997   89 min       Drama, Family, Sport          8.3   
98           1982   95 min      Drama, Fantasy, Music          8.1   
116          1920   76 min   Fantasy, Horror, Mystery          8.1   
146          1990  181 min  Adventure, Drama, Western          8.0   
152          1966  131 min                      Drama          8.0   
240          1993  118 min                      Drama          7.8   
243          1993   76 min   Animation, Action, Crime          7.8   
244          1989  108 min  Adventure, Comedy, Sci-Fi          7.8   
286          2002  116 min   Adventure, Comedy, Drama          7.7   
311          2007  140 min    Biography, Drama, Music          7.6   
326          1995  121 min  Adventure, Drama, Fantasy          7.6   

             Director               Star1                   Star2  \
8      Jonathan Demme        Jodie Foster         Anthony Hopkins   
25    Stanley Kubrick       Peter Sellers         George C. Scott   
33       Majid Majidi  Mohammad Amir Naji  Amir Farrokh Hashemian   
98        Alan Parker          Bob Geldof    Christine Hargreaves   
116      Robert Wiene       Werner Krauss            Conrad Veidt   
146     Kevin Costner       Kevin Costner          Mary McDonnell   
152      Mike Nichols    Elizabeth Taylor          Richard Burton   
240  Lasse Hallström         Johnny Depp       Leonardo DiCaprio   
243     Kevin Altieri       Boyd Kirkland              Frank Paur   
244   Robert Zemeckis      Michael J. Fox       Christopher Lloyd   
286          Bob Gale       James Marsden             Gary Oldman   
311     Olivier Dahan    Marion Cotillard           Sylvie Testud   
326      Jim Jarmusch         Johnny Depp             Gary Farmer   

                  Star3                   Star4  No_of_Votes        Gross  \
8    Lawrence A. Bonney            Kasi Lemmons      1270197  130,742,922   
25      Sterling Hayden             Keenan Wynn       450474      275,902   
33       Bahare Seddiqi  Nafise Jafar-Mohammadi        65341      933,933   
98      James Laurenson           Eleanor David        76081   22,244,207   
116     Friedrich Feher             Lil Dagover        57428          NaN   
146       Graham Greene         Rodney A. Grant       240266  184,208,848   
152        George Segal            Sandy Dennis        68926          NaN   
240      Juliette Lewis        Mary Steenburgen       215034    9,170,214   
243            Dan Riba           Eric Radomski        43690    5,617,391   
244        Lea Thompson        Thomas F. Wilson       481918  118,500,000   
286        Kurt Russell          Matthew Edison        29999          NaN   
311     Pascal Greggory      Emmanuelle Seigner        82781   10,301,706   
326      Crispin Glover         L

I only have 13 nan values out of 326, which i will fix manually:

In [27]:
df_before_filmaff.loc[df_before_filmaff["No_of_Votes"] ==  1270197, "Filmaffinity link"] = "https://www.filmaffinity.com/es/film768790.html"
df_before_filmaff.loc[df_before_filmaff["Spanish name"] == "¿Teléfono rojo? Volamos hacia Moscú", "Filmaffinity link"] = "https://www.filmaffinity.com/es/film479847.html"
df_before_filmaff.loc[df_before_filmaff["Spanish name"] == "Children of Heaven", "Filmaffinity link"] = "https://www.filmaffinity.com/es/film291595.html"
df_before_filmaff.loc[df_before_filmaff["Spanish name"] == "Pink Floyd: The Wall", "Filmaffinity link"] = "https://www.filmaffinity.com/es/film352368.html"
df_before_filmaff.loc[df_before_filmaff["Spanish name"] == "El gabinete del Dr. Caligari", "Filmaffinity link"] = "https://www.filmaffinity.com/es/film446167.html"
df_before_filmaff.loc[df_before_filmaff["No_of_Votes"] ==  240266, "Filmaffinity link"] = "https://www.filmaffinity.com/es/film239359.html"
df_before_filmaff.loc[df_before_filmaff["Spanish name"] == "¿Quién teme a Virginia Woolf?", "Filmaffinity link"] = "https://www.filmaffinity.com/es/film969127.html"
df_before_filmaff.loc[df_before_filmaff["Spanish name"] == "¿A quién ama Gilbert Grape?", "Filmaffinity link"] = "https://www.filmaffinity.com/es/film984814.html"
df_before_filmaff.loc[df_before_filmaff["No_of_Votes"] ==  43690, "Filmaffinity link"] = "https://www.filmaffinity.com/es/film526741.html"
df_before_filmaff.loc[df_before_filmaff["Spanish name"] == "Regreso al futuro. Parte II", "Filmaffinity link"] = "https://www.filmaffinity.com/es/film775323.html"
df_before_filmaff.loc[df_before_filmaff["Spanish name"] == "Interstate 60: Episodios de carretera", "Filmaffinity link"] = "https://www.filmaffinity.com/es/film239328.html"
df_before_filmaff.loc[df_before_filmaff["No_of_Votes"] ==  82781, "Filmaffinity link"] = "https://www.filmaffinity.com/es/film228749.html"
df_before_filmaff.loc[df_before_filmaff["No_of_Votes"] ==  90442, "Filmaffinity link"] = "https://www.filmaffinity.com/es/film711519.html"

all good:

In [28]:
df_before_filmaff[df_before_filmaff['Filmaffinity link'].isna()]

Empty DataFrame
Columns: [Series_Title, Released_Year, Runtime, Genre, IMDB_Rating, Director, Star1, Star2, Star3, Star4, No_of_Votes, Gross, Spanish name, Filmaffinity link]
Index: []

In [36]:
len(df_before_filmaff)

337

Lets get the stars and total voters in the filmaffinity base:

In [34]:
df_before_filmaff["Filmaffinity score"] = df_before_filmaff["Filmaffinity link"].apply(get_note_filmaffinity)

In [35]:
df_before_filmaff[df_before_filmaff['Filmaffinity score'].isna()]

Series_Title Released_Year  Runtime  \
15                   Vikram Vedha          2017  147 min   
30                      Incendies          2010  131 min   
35                   Idi i smotri          1985  142 min   
41                     Chhichhore          2019  143 min   
43               K.G.F: Chapter 1          2018  156 min   
50               Paan Singh Tomar          2012  135 min   
124                       Yip Man          2008  106 min   
137          Yeopgijeogin geunyeo          2001  137 min   
160                   Dil Bechara          2020  101 min   
204  M.S. Dhoni: The Untold Story          2016  184 min   
239            Breaking the Waves          1996  159 min   
260                   Kai po che!          2013  130 min   

                        Genre  IMDB_Rating          Director  \
15       Action, Crime, Drama          8.4           Gayatri   
30        Drama, Mystery, War          8.3  Denis Villeneuve   
35       Drama, Thriller, War          8.3       Elem Klimov   
41              Comedy, Drama          8.2     Nitesh Tiwari   
43              Action, Drama          8.2    Prashanth Neel   
50   Action, Biography, Crime          8.2  Tigmanshu Dhulia   
124  Action, Biography, Drama          8.0        Wilson Yip   
137    Comedy, Drama, Romance          8.0    Jae-young Kwak   
160    Comedy, Drama, Romance          7.9    Mukesh Chhabra   
204   Biography, Drama, Sport          7.8     Neeraj Pandey   
239                     Drama          7.8    Lars von Trier   
260              Drama, Sport          7.7   Abhishek Kapoor   

                    Star1                        Star2  \
15                Pushkar                     Madhavan   
30           Lubna Azabal  Mélissa Désormeaux-Poulin   
35     Aleksey Kravchenko                Olga Mironova   
41   Sushant Singh Rajput              Shraddha Kapoor   
43                   Yash              Srinidhi Shetty   
50            Irrfan Khan                   Mahie Gill   
124            Donnie Yen                    Simon Yam   
137          Tae-Hyun Cha                  Jun Ji-Hyun   
160  Sushant Singh Rajput               Sanjana Sanghi   
204  Sushant Singh Rajput                 Kiara Advani   
239          Emily Watson           Stellan Skarsgård   
260             Amit Sadh         Sushant Singh Rajput   

                      Star3               Star4  No_of_Votes      Gross  \
15         Vijay Sethupathi    Shraddha Srinath        28401        NaN   
30           Maxim Gaudette       Mustafa Kamel       150023  6,857,096   
35   Liubomiras Laucevicius     Vladas Bagdonas        59056        NaN   
41             Varun Sharma             Prateik        33893    898,575   
43         Ramachandra Raju        Archana Jois        36680        NaN   
50             Rajesh Abhay  Hemendra Dandotiya        33237     39,567   
124            Siu-Wong Fan         Ka Tung Lam       211427        NaN   
137              In-mun Kim        Song Wok-suk        45403        NaN   
160              Sahil Vaid  Saswata Chatterjee       111478        NaN   
204             Anupam Kher        Disha Patani        40416  1,782,795   
239        Katrin Cartlidge      Jean-Marc Barr        62428  4,040,691   
260           Rajkummar Rao         Amrita Puri        32628  1,122,527   

                     Spanish name  \
15                   Vikram Vedha   
30                      Incendios   
35                        Masacre   
41                     Chhichhore   
43               K.G.F: Chapter 1   
50               Paan Singh Tomar   
124                        Ip Man   
137                 My Sassy Girl   
160                   Dil Bechara   
204  M.S. Dhoni: The Untold Story   
239            Breaking the Waves   
260                   Kai po che!   

                                                                 Filmaffinity link  \
15                   https://www.filmaffinity.com/es/search.php?stext=vikram+vedha   
30                    

## CAREEEEE!!! 
The ones with nan value don't have a note in filmaffinity, so im gona drop them becouse they are useless to me

In [37]:
df_before_filmaff = df_before_filmaff.dropna(subset=['Filmaffinity score'])

In [38]:
df_before_filmaff[df_before_filmaff['Filmaffinity score'].isna()]

Empty DataFrame
Columns: [Series_Title, Released_Year, Runtime, Genre, IMDB_Rating, Director, Star1, Star2, Star3, Star4, No_of_Votes, Gross, Spanish name, Filmaffinity link, Filmaffinity score]
Index: []

Lets do the same with the total number of votes

In [42]:
df_before_filmaff["Filmaffinity votes"] = df_before_filmaff["Filmaffinity link"].apply(get_votes)

C:\Users\lenovo\AppData\Local\Temp\ipykernel_3092\1715886562.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_before_filmaff["Filmaffinity votes"] = df_before_filmaff["Filmaffinity link"].apply(get_votes)


In [43]:
df_before_filmaff[df_before_filmaff['Filmaffinity votes'].isna()]

Empty DataFrame
Columns: [Series_Title, Released_Year, Runtime, Genre, IMDB_Rating, Director, Star1, Star2, Star3, Star4, No_of_Votes, Gross, Spanish name, Filmaffinity link, Filmaffinity score, Filmaffinity votes]
Index: []

There are no nan values in the row.

In [56]:
df_before_filmaff["Runtime"] = df["Runtime"]

C:\Users\lenovo\AppData\Local\Temp\ipykernel_3092\370503871.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_before_filmaff["Runtime"] = df["Runtime"]


In [57]:
df_before_filmaff

Series_Title  \
0                                                  The Godfather: Part II   
1                                                        Schindler's List   
2                                         Il buono, il brutto, il cattivo   
3                          Star Wars: Episode V - The Empire Strikes Back   
4                                                            Gisaengchung   
5                                                          Cidade de Deus   
6                                           Sen to Chihiro no kamikakushi   
7                                                          The Green Mile   
8                                                The Silence of the Lambs   
9                                                    Shichinin no samurai   
10                                                       The Intouchables   
11                                                           The Prestige   
12                                                     American History X   
13                                                  Nuovo Cinema Paradiso   
14                                              Ayla: The Daughter of War   
16                                                         Kimi no na wa.   
17                                      Spider-Man: Into the Spider-Verse   
18                                                      Avengers: Endgame   
19                                                 Avengers: Infinity War   
20                                                               3 Idiots   
21                                                       Taare Zameen Par   
22                                                               Oldeuboi   
23                                                          Mononoke-hime   
24                                            Once Upon a Time in America   
25   Dr. Strangelove or: How I Learned to Stop Worrying and Love the Bomb   
26                                                     The Great Dictator   
27                                                                Tumbbad   
28                                                              Andhadhun   
29                                                Jodaeiye Nader az Simin   
31                                                         Babam ve Oglum   
32                                                      Good Will Hunting   
33                                                      Bacheha-Ye aseman   
34                                                      Full Metal Jacket   
36                                                     A Clockwork Orange   
37                                                  2001: A Space Odyssey   
38                                                     North by Northwest   
39                                                    Singin' in the Rain   
40                                                    Ladri di biciclette   
42                                               Uri: The Surgical Strike   
44                                                             Green Book   
45                              Three Billboards Outside Ebbing, Missouri   
46                                                                 Talvar   
47                                            Baahubali 2: The Conclusion   
48                                                            Mandariinid   
49                                                     Gangs of Wasseypur   
51                                                 El secreto de sus ojos   
52                                                         Chak De! India   
53                                                        Pan's Labyrinth   
54                                                         V for Vendetta   
55                                                        Rang De Basanti   
56                                                 Swades: We, the People   
57                                                   Haur

### Export

Export the clean data frame to a new CSV file

In [45]:
df_before_filmaff.to_csv("src/clean_data_filmaff_imdb.csv",index=False)